cli/flox-rust-sdk/src/models/manifest/composite/shallow.rs

```diff
@@ -0,0 +1,499 @@
+        Ok(merged)
+    }
+
+    fn merge_services(
+        low_priority: &ManifestServices,
+        high_priority: &ManifestServices,
+    ) -> Result<ManifestServices, MergeError> {
+        let merged = map_union(low_priority.inner(), high_priority.inner());
+        Ok(ManifestServices(merged))
+    }
+
+    fn merge_build(
+        low_priority: &ManifestBuild,
+        high_priority: &ManifestBuild,
+    ) -> Result<ManifestBuild, MergeError> {
+        let merged = map_union(low_priority.inner(), high_priority.inner());
+        Ok(ManifestBuild(merged))
+    }
+
+    fn merge_containerize(
+        low_priority: Option<&ManifestContainerize>,
+        high_priority: Option<&ManifestContainerize>,
+    ) -> Result<Option<ManifestContainerize>, MergeError> {
+        match (low_priority, high_priority) {
+            (None, None) => Ok(None),
+            (Some(containerize_lp), None) => Ok(Some(containerize_lp.clone())),
+            (None, Some(containerize_hp)) => Ok(Some(containerize_hp.clone())),
+            (
+                Some(ManifestContainerize { config: cfg_lp }),
+                Some(ManifestContainerize { config: cfg_hp }),
+            ) => {
+                let merged =
+                    deep_merge_optional_containerize_config(cfg_lp.as_ref(), cfg_hp.as_ref());
+                Ok(Some(ManifestContainerize { config: merged }))
+            },
+        }
+    }
+
+    fn merge(
+        &self,
+        low_priority: &Manifest,
+        high_priority: &Manifest,
+    ) -> Result<Manifest, MergeError> {
+        let manifest = Manifest {
+            version: Self::merge_version(&low_priority.version, &high_priority.version)?,
+            install: Self::merge_install(&low_priority.install, &high_priority.install)?,
+            vars: Self::merge_vars(&low_priority.vars, &high_priority.vars)?,
+            hook: Self::merge_hook(&low_priority.hook, &high_priority.hook)?,
+            profile: Self::merge_profile(&low_priority.profile, &high_priority.profile)?,
+            options: Self::merge_options(&low_priority.options, &high_priority.options)?,
+            services: Self::merge_services(&low_priority.services, &high_priority.services)?,
+            build: Self::merge_build(&low_priority.build, &high_priority.build)?,
+            containerize: Self::merge_containerize(
+                low_priority.containerize.as_ref(),
+                high_priority.containerize.as_ref(),
+            )?,
+        };
+
+        Ok(manifest)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use flox_test_utils::proptest::btree_maps_overlapping_keys;
+    use pretty_assertions::assert_eq;
+    use proptest::prelude::*;
+
+    use super::*;
+    use crate::models::manifest::typed::{
+        Allows,
+        ManifestBuildDescriptor,
+        ManifestContainerizeConfig,
+        ManifestPackageDescriptor,
+        ManifestServiceDescriptor,
+        SemverOptions,
+    };
+
+    proptest! {
```

Comment:
For most of the proptest tests, i think summarizing the test into a doc comment would already make it easier to understand the tests, i.e. prime you as to what to expect rather than going through and understanding the code.

Reply:
Fixed